axon-mongo-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/mongo/springboot/autoconfig/MongoAutoConfiguration.java

```diff
@@ -0,0 +1,139 @@
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.mongodb.MongoDatabaseFactory;
+import org.springframework.data.mongodb.MongoTransactionManager;
+
+import static java.util.Objects.isNull;
+
+@Configuration
+@EnableConfigurationProperties(AxonMongoProperties.class)
+@AutoConfigureAfter(name = {
+        "org.axonframework.springboot.autoconfig.TransactionAutoConfiguration",
+        "org.axonframework.springboot.autoconfig.JdbcAutoConfiguration"
+})
+public class MongoAutoConfiguration {
+
+    private final AxonMongoProperties axonMongoProperties;
+
+    public MongoAutoConfiguration(AxonMongoProperties axonMongoProperties) {
+        this.axonMongoProperties = axonMongoProperties;
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public MongoTransactionManager mongoTransactionManager(MongoDatabaseFactory factory) {
+        return new MongoTransactionManager(
+                factory,
+                TransactionOptions.builder().writeConcern(WriteConcern.MAJORITY).build()
+        );
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public MongoTemplate axonMongoTemplate(MongoDatabaseFactory factory) {
+        String databaseName = axonMongoProperties.getDatabaseName();
+        if (isNull(databaseName)) {
+            return SpringMongoTemplate.builder()
+                                      .factory(factory)
+                                      .build();
+        } else {
+            return SpringMongoTemplate.builder()
+                                      .factory(factory)
+                                      .databaseName(databaseName)
+                                      .build();
+        }
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public TransactionManager transactionManager(MongoTransactionManager manager) {
+        return new SpringMongoTransactionManager(manager);
+    }
+
+    @Bean("tokenStore")
+    @ConditionalOnMissingBean(TokenStore.class)
+    @ConditionalOnProperty(value = "axon.mongo.token-store.enabled", matchIfMissing = true)
+    public TokenStore tokenStore(
+            MongoTemplate mongoTemplate,
+            TransactionManager transactionManager,
+            Serializer serializer
+    ) {
+        return MongoTokenStore.builder()
+                              .mongoTemplate(mongoTemplate)
+                              .transactionManager(transactionManager)
+                              .serializer(serializer)
+                              .build();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean({EventStorageEngine.class, EventBus.class})
+    @ConditionalOnProperty(value = "axon.mongo.event-store.enabled")
+    public EventStorageEngine eventStorageEngine(
```

Comment:
Might be nice if users can define the `StorageStrategy` they want to use. Perhaps we can add this to the `AxonMongoProperties` too?

Reply:
Added it, but since it needs a class instance I'm not 100% sure the current implementation is good.

Reply:
Ow, I wasn't overly clear I guess. I thought I shared (so, my bad) that we could do this through an enumeration. Thus, where the enumeration dictates the default implementation, which users can override by providing a `StorageStrategy` bean.

On the other end, now that I see your implementation, perhaps we shouldn't do this through the `AxonMongoProperties` file... Just providing a bean creation method in this auto configuration may be sufficient.

What do you think, @gklijs?

Reply:
After some more thought, changed to a bean on the same property as the engine. So they can create another storage strategy bean if they want to.